### PR TITLE
feat(tsoa): `bring tsoa`

### PR DESCRIPTION
**Pull request by ekeren**

### CODE REVIEW

File: tsoa/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# tsoa
+
+A Wing library for working with TSOA projects.
```
Comment (Chriscbr) [inline tsoa/README.md:3]: what is TSOA? Tiny Sandwiches On Airplanes? 😄

File: tsoa/lib/app.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+const express = require("express");
```
Comment (eladb) [inline tsoa/lib/app.js:1]: change to `.ts` maybe?

File: tsoa/lib/templates/express.hbs [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                            expressAuthenticationRecasted(request, name, secMethod[name], response)
+                                .catch(pushAndRethrow)
+                        );
+                    }
+                }
+            }
+
+            // WARNING: This file was auto-generated with tsoa. Please do not modify it. Re-run tsoa to re-generate this file: https://github.com/lukeautry/tsoa
+
+            try {
```
Comment (eladb) [inline tsoa/lib/templates/express.hbs:170]: do we need to check this in?

File: tsoa/package.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    "url": "https://github.com/winglang/winglibs.git",
+    "directory": "tsoa"
+  },
+  "wing": {
+    "platforms": [
+      "sim"
+    ]
+  },
+  "license": "MIT",
+  "dependencies": {
```
Comment (eladb) [inline tsoa/package.json:20]: should all be peers right @MarkMcCulloh ?

File: tsoa/test/usersController.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	Body,
+	Controller,
+	Get,
+	Path,
+	Post,
+	Query,
+	Route,
+	SuccessResponse,
+} from "tsoa";
+// import type { User } from "./user.js";
```
Comment (eladb) [inline tsoa/test/usersController.ts:11]: delete commented code?

File: tsoa/lib/lib.w [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+      return inflight () => {
+        res.close();
+      };
+    });
+
+    this.setUi();
+  }
+
+  setUi() {
```
Comment (eladb) [inline tsoa/lib/lib.w:54]: nit:

```suggestion
  addUi() {
```

File: tsoa/lib/lib.w [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  inflight close(): void;
+}
+
+struct SpecProps {
+  outputDirectory: str?;
+  specVersion: num?;
+}
+
+pub struct ServiceProps {
+  entryFile: str?;
```
Comment (eladb) [inline tsoa/lib/lib.w:17]: add some docstrings?

@yoav-steinberg @Chriscbr what are the prospects for docstring support?

File: tsoa/lib/lib.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+      } : undefined,
+    };
+
+    const routeOptions = {
+      "entryFile": options.entryFile ? join(basedir, options.entryFile) : "./app.js",
+      "noImplicitAdditionalProperties": "throw-on-extras",
+      "controllerPathGlobs": options.controllerPathGlobs.map((path) => join(basedir, path)),
+      "routesDir": join(basedir, options.routesDir),
+      "bodyCoercion": false,
+      "middlewareTemplate": join(basedir, "./templates/express.hbs"),
```
Comment (eladb) [inline tsoa/lib/lib.js:26]: nit:

```suggestion
      entryFile: options.entryFile ? join(basedir, options.entryFile) : "./app.js",
      noImplicitAdditionalProperties: "throw-on-extras",
      controllerPathGlobs: options.controllerPathGlobs.map((path) => join(basedir, path)),
      routesDir: join(basedir, options.routesDir),
      bodyCoercion: false,
      middlewareTemplate: join(basedir, "./templates/express.hbs"),
```

File: tsoa/lib/lib.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  try {
+    const specOptions = {
+      "entryFile": options.entryFile ? join(basedir, options.entryFile) : "./app.js",
+      "noImplicitAdditionalProperties": "throw-on-extras",
+      "controllerPathGlobs": options.controllerPathGlobs.map((path) => join(basedir, path)),
+      "outputDirectory": join(basedir, options.outputDirectory),
+      "spec": options.spec ? {
+        "outputDirectory": options.spec.outputDirectory ? join(basedir, options.spec.outputDirectory) : undefined,
+        "specVersion": options.spec.specVersion ?? 3
+      } : undefined,
```
Comment (eladb) [inline tsoa/lib/lib.js:17]:
```suggestion
      entryFile: options.entryFile ? join(basedir, options.entryFile) : "./app.js",
      noImplicitAdditionalProperties: "throw-on-extras",
      controllerPathGlobs: options.controllerPathGlobs.map((path) => join(basedir, path)),
      outputDirectory: join(basedir, options.outputDirectory),
      spec: options.spec ? {
        outputDirectory: options.spec.outputDirectory ? join(basedir, options.spec.outputDirectory) : undefined,
        specVersion: options.spec.specVersion ?? 3
      } : undefined,
```

File: tsoa/lib/lib.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
+const {
+  generateRoutes,
+  generateSpec,
+} = require("tsoa");
+const { join } = require("node:path");
+
+exports.startService = async (basedir, workdir, options) => {
+  try {
+    const specOptions = {
```
Comment (eladb) [inline tsoa/lib/lib.js:9]: would it make sense to expose these options to the user?

File: tsoa/lib.w [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+ */
+pub class Service {
+  pub url: str;
+  state: sim.State;
+  service: cloud.Service;
+
+  new(props: ServiceProps) {
+    let target = util.env("WING_TARGET");
+    if target != "sim" {
+      throw "Unsupported target {target}";
```
Comment (ekeren) [inline tsoa/lib.w:53]: @eladcon can you estimate how much effort is required to crest an tf-aws platform (that uses api gateway)?

File: tsoa/README.md (hunk 1 above)
Comment (Chriscbr) [reply tsoa/README.md]: neat! Maybe add the link to the README?

File: tsoa/test/usersController.ts (hunk 5 above)
Comment (ekeren) [inline tsoa/test/usersController.ts]:
```suggestion
```

**Review by staycoolcall911 — APPROVED** [review]

**Pull request merged**